core/dbt/parser/results.py

```diff
@@ -0,0 +1,191 @@
+from dataclasses import dataclass, field
+from typing import TypeVar, Dict, Mapping, Union, List
+
+from hologram import JsonSchemaMixin
+
+from dbt.contracts.graph.manifest import SourceFile, RemoteFile, FileHash
+from dbt.contracts.graph.parsed import (
+    ParsedNode, HasUniqueID, ParsedMacro, ParsedDocumentation, ParsedNodePatch,
+    ParsedSourceDefinition, ParsedAnalysisNode, ParsedHookNode, ParsedRPCNode,
+    ParsedModelNode, ParsedSeedNode, ParsedTestNode, ParsedSnapshotNode,
+)
+from dbt.contracts.util import Writable
+from dbt.exceptions import (
+    raise_duplicate_resource_name, raise_duplicate_patch_name,
+    CompilationException, InternalException
+)
+
+
+# Parsers can return anything as long as it's a unique ID
+ParsedValueType = TypeVar('ParsedValueType', bound=HasUniqueID)
+
+
+def _check_duplicates(value: HasUniqueID, src: Mapping[str, HasUniqueID]):
+    if value.unique_id in src:
+        raise_duplicate_resource_name(value, src[value.unique_id])
+
+
+ManifestNodes = Union[
+    ParsedAnalysisNode,
+    ParsedHookNode,
+    ParsedModelNode,
+    ParsedSeedNode,
+    ParsedTestNode,
+    ParsedSnapshotNode,
+    ParsedRPCNode,
+]
+
+
+@dataclass
+class ParseResult(JsonSchemaMixin, Writable):
+    vars_hash: FileHash
+    profile_hash: FileHash
+    project_hashes: Dict[str, FileHash]
+    nodes: Dict[str, ManifestNodes] = field(default_factory=dict)
+    sources: Dict[str, ParsedSourceDefinition] = field(default_factory=dict)
+    docs: Dict[str, ParsedDocumentation] = field(default_factory=dict)
+    macros: Dict[str, ParsedMacro] = field(default_factory=dict)
+    patches: Dict[str, ParsedNodePatch] = field(default_factory=dict)
+    files: Dict[str, SourceFile] = field(default_factory=dict)
+    disabled: Dict[str, List[ParsedNode]] = field(default_factory=dict)
```

Comment:
This needs to have information that `Manifest` does not, and is missing information that `Manifest` has - the possibilities are more constrained, as you can't have compiled entities, and we keep track of file paths to the various disabled nodes instead of a flattened list. I thought about making it a subclass but then there were annoying type issues